### PR TITLE
Add calculator.php

### PR DESCRIPTION
**Pull request by HLongNguyen0**

### CODE REVIEW

File: calculator.php [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+
+    public function __construct() {
+        do {
+            $this->total = readline("Enter 1 number : ");
+            if(is_numeric($this->total)) {
+                break;
+            }
+            echo "Invalid number\n";
+        } while(true);
```
Comment (yhabteab) [inline calculator.php:17]: A constructor shouldn't contain such a complicated logic. A constructor should just initialize the member fields if needed but nothing else. Please outsource this code into a separate method.

File: calculator.php [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+            break;
+        case $calc->getArr()[6]:
+            $calc->power();
+            break;
+
+    }
+    if ($calc->getOperator() != $calc->getArr()[5]) {  
+        $loop++;
+    }
+} while($calc->getOperator() !== $calc->getArr()[7]);
```
Comment (yhabteab) [inline calculator.php:120]: It should  also be possible to put all this stuff int a member method of the `Calc` class e.g `calculate()`. You would just use it like:
```php
$calc = new Calc();
$calc->calculate();
```

File: calculator.php [hunk 3]
```diff
@@ -0,0 +1,122 @@
+<?php
+
+class Calc {
+    private $total;
+    private $operator;
+    private $num;
+    private const OPERSTIONS = array("+", "-", "*", "/", "%", "!", "**", "=");
```
Comment (yhabteab) [inline calculator.php:7]: `OPERSTIONS`? I think `OPERATORS` would fit better as a name.

File: calculator.php [hunk 4]
```diff
@@ -0,0 +1,122 @@
+<?php
+
+class Calc {
+    private $total;
```
Comment (yhabteab) [inline calculator.php:4]: Think of getting away from this property.

File: calculator.php (hunk 1 above)
Comment (HLongNguyen0) [reply calculator.php:22]: I got rid of __constructor completely and added a new method setTotal()

File: calculator.php (hunk 2 above)
Comment (HLongNguyen0) [reply calculator.php]: I added a method calculate()

File: calculator.php (hunk 4 above)
Comment (HLongNguyen0) [reply calculator.php:8]: And here I'm not even sure how I can do this. Almost every method in my class uses $total property

Comment (yhabteab) [reply calculator.php:8]: Yeah, that's the point. They should just return their results and don't need to cache it.

Comment (HLongNguyen0) [reply calculator.php:8]: But I need to cache it to use it for next operations. I'm able to keep adding to $total as much as I want

**Comment by HLongNguyen0** [comment]
Can I merge this branch yet?

**Pull request closed without merging**